Watch sync and restart action (#18463)
docker/docs: content/compose/file-watch.md

Watch mode can be used with many different languages and frameworks.
The specific paths and rules will vary project to project, but the concepts remain the same. 

### Prerequisites

> To work properly `watch` relies on common executables, so be sure your service image will contain the following binaries:

Reviewer: ```suggestion
In order to work properly, `watch` relies on common executables. Make sure your service image contains the following binaries:
```
